tf2/src/cache.cpp

```diff
@@ -250,23 +250,38 @@ bool TimeCache::insertData(const TransformStorage & new_data)
 {
   const TimePoint latest_time = getLatestTimestamp();
 
+  // Always prune data. This (a) ensures we trim data we should not access, and
+  // (b) ensures we don't waste time iterating over items to be removed.
+  pruneList();
```

Comment:
I like doing this first.  It both improves performance for below, and also means that the list is always pruned, even if we are not going to insert the data.